api/host.go

```diff
@@ -5,65 +5,113 @@ import (
+		UnlockHash   types.UnlockHash   `json:"unlockhash"`
+		WindowSize   types.BlockHeight  `json:"windowsize"`
+
+		NumContracts     uint64         `json:"numcontracts"`
+		Revenue          types.Currency `json:"revenue"`
+		StorageRemaining int64          `json:"storageremaining"`
+		UpcomingRevenue  types.Currency `json:"upcomingrevenue"`
 	}
-	if err != nil {
-		writeError(w, err.Error(), http.StatusBadRequest)
-		return
+)
+
+// hostHandlerGET handles GET requests to the /host API endpoint.
+func (srv *Server) hostHandlerGET(w http.ResponseWriter, req *http.Request) {
+	settings := srv.host.Settings()
+	upcomingRevenue, revenue := srv.host.Revenue()
+	storageRemaining, numContracts := srv.host.Capacity()
+	hg := HostGET{
+		Collateral:   settings.Collateral,
+		IPAddress:    settings.IPAddress,
+		MaxDuration:  settings.MaxDuration,
+		MinDuration:  settings.MinDuration,
+		Price:        settings.Price,
+		TotalStorage: settings.TotalStorage,
+		UnlockHash:   settings.UnlockHash,
+		WindowSize:   settings.WindowSize,
+
+		NumContracts:     numContracts,
+		Revenue:          revenue,
+		StorageRemaining: storageRemaining,
+		UpcomingRevenue:  upcomingRevenue,
 	}
-	writeSuccess(w)
+	writeJSON(w, hg)
 }
 
-// hostConfigureHandler handles the API call to set the host configuration.
-func (srv *Server) hostConfigureHandler(w http.ResponseWriter, req *http.Request) {
-	// load current settings
-	config := srv.host.Info().HostSettings
-
-	// map each query string to a field in the host announcement object
+// hostHandlerPOST handles POST request to the /host API endpoint.
+func (srv *Server) hostHandlerPOST(w http.ResponseWriter, req *http.Request) {
+	// Map each query string to a field in the host settings.
+	settings := srv.host.Settings()
 	qsVars := map[string]interface{}{
-		"totalstorage": &config.TotalStorage,
-		"minfilesize":  &config.MinFilesize,
-		"maxfilesize":  &config.MaxFilesize,
-		"minduration":  &config.MinDuration,
-		"maxduration":  &config.MaxDuration,
-		"windowsize":   &config.WindowSize,
-		"price":        &config.Price,
-		"collateral":   &config.Collateral,
+		"collateral":   &settings.Collateral,
+		"maxduration":  &settings.MaxDuration,
+		"minduration":  &settings.MinDuration,
+		"price":        &settings.Price,
+		"totalstorage": &settings.TotalStorage,
+		"windowsize":   &settings.WindowSize,
 	}
 
-	any := false
+	// Iterate through the query string and replace any fields that have been
+	// altered.
 	for qs := range qsVars {
-		// only modify supplied values
-		if req.FormValue(qs) != "" {
+		if req.FormValue(qs) != "" { // skip empty values
 			_, err := fmt.Sscan(req.FormValue(qs), qsVars[qs])
 			if err != nil {
 				writeError(w, "Malformed "+qs, http.StatusBadRequest)
 				return
 			}
-			any = true
 		}
 	}
-	if !any {
-		writeError(w, "No valid configuration fields specified", http.StatusBadRequest)
-		return
+	srv.host.SetSettings(settings)
+	writeSuccess(w)
```

Comment:
Seemed like an odd corner case to drop an error if there were no changes.

I don't feel strongly about it either way, just seemed like it added confusing code to handle an edge case I didn't feel strongly about.

I'll bring it back if you want.